packages/ckeditor5-dev-changelog/src/generatechangelog.ts

```diff
@@ -25,20 +25,40 @@ import { getRepositoryUrl } from './utils/external/getrepositoryurl.js';
 import { getNewChangelog } from './utils/getnewchangelog.js';
 import { removeChangesetFiles } from './utils/removechangesetfiles.js';
 import { removeScope } from './utils/removescope.js';
+import { InternalError } from './errors.js';
 
 export async function generateChangelog(
-  config: RepositoryConfig & GenerateChangelog & { noWrite?: false | undefined }
+	config: RepositoryConfig & GenerateChangelog & { noWrite?: false | undefined }
 ): Promise<void>;
 
 export async function generateChangelog(
-  config: RepositoryConfig & GenerateChangelog & { noWrite: true }
+	config: RepositoryConfig & GenerateChangelog & { noWrite: true }
 ): Promise<string>;
 
+/**
+ * Wrapper function that provides error handling for the changelog generation process.
+ */
+export async function generateChangelog(
+	options: RepositoryConfig & GenerateChangelog
+): Promise<string | void> { // eslint-disable-line @typescript-eslint/no-invalid-void-type
+	try {
+		return await main( options );
+	} catch ( error ) {
+		if ( error instanceof InternalError ) {
+			console.error( chalk.red( 'Error: ' + error.message ) );
+
+			process.exit( 1 );
+		} else {
+			throw error;
+		}
```

Comment:
```suggestion
		if ( !( error instanceof InternalError ) ) {
			throw error;
		}
	
		console.error( chalk.red( 'Error: ' + error.message ) );
		process.exit( 1 );
```

One fewer `{}` :D

Reply:
Also consider borrowing the `removeWhitespace()` util from `build-tools`. With it, you can write this error message in string literal, and it will remove all whitespace characters after every new line. It looks much cleaner.

https://github.com/ckeditor/ckeditor5-dev/blob/a9bdc6572d0caa33b22608a722847f960e4267d1/packages/ckeditor5-dev-build-tools/src/plugins/translations.ts#L14-L21

Reply:
@filipsobol using `removeWhitespace` would be nice, but that would require to add `ckeditor5-dev-build-tools` as a dependency. Are we sure we want to do that just to use one util?

Reply:
@pomek I simplified the global error handling in [87252be](https://github.com/ckeditor/ckeditor5-dev/pull/1132/commits/87252beaf98bdd5010efcb11a5a346d38fc713d4). I left the message in `InternalError`, because I think this is not a bad solution. We could change the name of the error to e.g. `InvalidChangesets` or something similar. But in the current approach, we could add as many error types as we want, without increasing the size of the entry script `generateChangelog.ts`.